src/js/modules/Product.js

```diff
@@ -0,0 +1,111 @@
+import { template } from "./product-template.js";
+import Wish from "./wish.js";
+import ProductServices from "../services/sharedproduct.service.js";
+
+class Product {
+  startIndex = 0;
+  pageSize = 3;
+  currentProducts = [];
+  productListElement = document.getElementById("product-list");
+  numberProducts = 0;
+  numberProductsContainer = document.getElementById("numberProducts");
+
+  constructor() {
+    this.loadProducts();
+    this.initEvents();
+  }
+
+  loadProducts = async () => {
+    try {
+      await ProductServices.fetchProducts();
+      this.numberProductsContainer.innerHTML = ProductServices.productsCount;
+      this.loadNewProductsToRender();
+    } catch (err) {
+      console.log(err);
+    }
+  };
+
+  displayProducts(products = []) {
+    let existingCart = ProductServices.getCartProducts();
+  
+    const promise = new Promise((resolve, reject) => {
+        products.forEach((product) => {
+        this.productListElement.appendChild(template(product));
+        });
+        this.initAddToCartEvent();
+        ProductServices.setCartValues(existingCart);
+        resolve(true);
+    })
+    promise.then(()=>{
+      Wish.initWish()
+    })
+  }
+  loadNewProductsToRender(startIndex = 0, pageSize = this.pageSize) {
+    const loadedProducts = ProductServices.getProducts(startIndex, pageSize);
+    if (loadedProducts.length === 0 || loadedProducts.length < pageSize) {
+      const button = document.getElementById("load-more");
+      button.classList.add("hide");
+    } else {
+      this.currentProducts = [...this.currentProducts, ...loadedProducts];
+    }
+
+    // save new products
+    ProductServices.saveProducts(this.currentProducts);
+    // Display products
+    this.displayProducts(loadedProducts);
+    //
+  }
+
+  loadMore() {
+    // Change page index
+    this.startIndex = this.startIndex + this.pageSize;
+    this.loadNewProductsToRender(this.startIndex);
+  }
+
+  initAddToCartEvent() {
+    let button = this.productListElement.querySelectorAll(".js-add-cart");
+    const self = this;
+    button.forEach(function (addBtn) {
+      addBtn.addEventListener("click", function (e) {
+        let productId = Number(e.currentTarget.getAttribute("data-id"));
+        const targetProduct = ProductServices.allProducts.find(
```

Comment:
get product should be inside `addToCartProducts `instead, we only get `productId `here